aws_lambda_powertools/utilities/parameters/dynamodb.py

```diff
@@ -152,15 +158,19 @@ def __init__(
         endpoint_url: Optional[str] = None,
         config: Optional[Config] = None,
         boto3_session: Optional[boto3.session.Session] = None,
+        boto3_client: Optional["DynamoDBServiceResource"] = None,
```

Comment:
Agreed on the docs, I will make those changes tomorrow (docs, not param name) and merge it.

Reply:
thanks clarity always helps.

Reply:
overall for `DynamoDBProvider` the resource client so that you can set the endpoint url seems redundant when you already have `endpoint_url` as a parameter. 

With 8 parameters now, this is borderline code smell.

Reply:
@heitorlessa  the docs definitely helps! It might be a good general call out to recommend people use boto3 session where possible